dex/src/state.rs

```diff
@@ -2499,6 +2637,8 @@ impl State {
             fee_tier,
         } = args;
 
+        let open_orders_mut = open_orders.deref_mut();
```

Comment:
This change is required so that an open orders PDA can be the authority of itself. On CPI, `solana_program::program::invoke` will try to borrow the refcell, and so this borrow must be dropped prior to CPI.